os/pm/pm_wakehandler.c

```diff
@@ -56,7 +56,7 @@ void pm_wakehandler(clock_t missing_tick, pm_wakeup_reason_code_t wakeup_src)
 {
 	irqstate_t flags = enter_critical_section();
 #ifdef CONFIG_PM_METRICS
-	pm_metrics_update_wakehandler(missing_tick, wakeup_src);
+	pm_metrics_update_wakehandler((uint32_t)missing_tick, wakeup_src);
```

Comment:
why do missing_tick is typecased to uint32_t?

Reply:
The reason is `missing_tick` is of type `uint64` while `pm_metrics` support only `uint32`.

Reply:
We need to organize the type in pm_metrics.

please check and let me know the maximum time supported by pm_metrics.

You can limit all actions to the maximum time and change the data type correctly.

Reply:
@ewoodev We can scale `pm_metrics` to `UINT64_MAX`

Reply:
I mean, currently our pm_metrics use pm_sleep. This is the maximum int max of delay.
Therefore, the pm_metrics input is int max tick.

is it necessary for the count and ticks of pm_metrics to maintain int max tick?

Reply:
Yes, but it is highly unlikely board will sleep continuously for `INT32_MAX` i.e. `24 days`.Since we have `UART` and `Wifi` Interrupts, it guarantee the board will wakeup at least one time within 24 days sleep. So we can scale our `pm_metrics` to `UINT64_MAX`.